src/silly_sprites/opengl/buffer_object.cpp

```diff
@@ -0,0 +1,97 @@
+#include "buffer_object.hpp"
+
+namespace sly::gl {
+    void BufferObject::init() {
+        generate_vertex_array();
+        generate_vertex_buffer();
+        generate_element_buffer();
+
+        glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), static_cast<void*>(0));
+        glEnableVertexAttribArray(0);
```

Comment:
Is this always `0`? Shouldn't this name(*) come from some OpenGL call? But I'm not sure here.

(*)OpenGL calls those id values "names" (just in case you didn't know already).

Reply:
should i also remove the 3?
if i do i have to make the drawing mode settable too right?

will we draw only triangles?

Reply:
This will need to be more flexible later on during development, since we will require more vertex attributes (e.g. position [vec3], color [rgba, either 4xfloat or 4xuint8], uv-coordinates [vec2]).